core/src/main/java/org/jruby/truffle/runtime/core/RubyArray.java

```diff
@@ -244,8 +216,11 @@ public int getSize() {
     }
 
     public void setSize(int size) {
+        assert verifyStore(this.store, size);
```

Comment:
This should also be a general style guide. Put long assertion code in extra helpers that return boolean "true" and are invoked themselves with "assert methodCall". Again improves readability but also startup/warmup.